fix(auth): export MISSING_PASSWORD in AuthErrorCodes

Navya-arugonda opened the pull request:
**Summary**
Export `MISSING_PASSWORD: 'auth/missing-password'` via `AuthErrorCodes` in @firebase/auth.

**Why**
The SDK can throw `auth/missing-password`, but it wasn’t exposed via `AuthErrorCodes`, forcing developers to hardcode the string.

**Tests**
Adds a Node unit test asserting:
`AuthErrorCodes.MISSING_PASSWORD === 'auth/missing-password'`.

**Fixes**
Fixes #9270 

**CLA**
I have signed the Google Contributor License Agreement.

dlarocque commented on packages/auth/src/core/errors.ts:
```diff
@@ -603,5 +603,6 @@ export const AUTH_ERROR_CODES_MAP_DO_NOT_USE_INTERNALLY = {
   INVALID_REQ_TYPE: 'auth/invalid-req-type',
-  INVALID_HOSTING_LINK_DOMAIN: 'auth/invalid-hosting-link-domain'
+  INVALID_HOSTING_LINK_DOMAIN: 'auth/invalid-hosting-link-domain',
+  MISSING_PASSWORD: 'auth/missing-password'
```
The properties of this object closely mirror the internal `AuthErrorCode` enum above. To keep the ordering consistent, can you move this new property below `MISSING_PHONE_NUMBER` on line 562?
Here's the ordering in the internal enum:
https://github.com/firebase/firebase-js-sdk/blob/ccbf7ba36fa9b10c100296489f8be265c8f83e41/packages/auth/src/core/errors.ts#L89-L95

dlarocque commented on packages/auth/tsconfig.json:
```diff
@@ -1,7 +1,11 @@
   "extends": "../../config/tsconfig.base.json",
   "compilerOptions": {
-    "outDir": "dist"
+    "outDir": "dist",
+    "types": ["mocha", "chai"],
+    "esModuleInterop": true,
+    "moduleResolution": "node",
+    "resolveJsonModule": true
```
Why are these changes necessary?

dlarocque commented:
When we make additions to public APIs, we have to update our reference documentation here on this website: https://firebase.google.com/docs/reference. The documentation hosted on that website is generated from the code in our SDK, and stored in the `docs-devsite/` directory.

Since you made a public API addition, you need to generate the documentation for this new API. You can do this by running the `yarn docgen:all` script and committing the changes.

Navya-arugonda commented:
Thanks for the review!
I’ve moved MISSING_PASSWORD below MISSING_PHONE_NUMBER, reverted the tsconfig.json changes, and removed the unnecessary test file.
Please let me know if any further adjustments are needed.

dlarocque commented on packages/auth/tsconfig.json:
```diff
@@ -3,5 +3,6 @@
   "compilerOptions": {
     "outDir": "dist"
   },
+  "types": ["jasmine"],
```
Please also revert this addition since it is not necessary.

Navya-arugonda replied:
removed the line and updated the code.

Navya-arugonda commented:
Added the generated auth.api.md via yarn build.
